src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -91,6 +91,80 @@ public class TestExtensions
     /// </summary>
     internal bool AreDataCollectorsCached { get; set; }
 
+    /// <summary>
+    /// Merge two extension dictionaries.
+    /// </summary>
+    /// 
+    /// <param name="first">First extension dictionary.</param>
+    /// <param name="second">Second extension dictionary.</param>
+    /// 
+    /// <returns>
+    /// A dictionary representing the merger between the two input dictionaries.
+    /// </returns>
+    internal static Dictionary<string, HashSet<string>> MergeDictionaries(
```

Comment:
[minor] Now that this is pure function, maybe the name shall be `MakeMergedDictionary` or `CreateMergedDictionary`. Or maybe we should even go with a more functional name `CreateMergedDiscoveredExtensions`. WDYT?

Reply:
I think we can go with `CreateMergedDictionary`. Going with `CreateMergedDiscoveredExtensions` would throw us back to the discussion where we debated a more generic naming.